aws/s3_test.go

```diff
@@ -132,332 +161,331 @@ func (b TestS3Bucket) addObject(s SetupInfo, fileName string, fileBody string) e
 	if err != nil {
 		return err
 	}
-
 	return nil
 }
 
-// createSession creates a new session and returns it.
-func createSession(region string) (*session.Session, error) {
-	if region == "" {
-		var err error
-		region, err = getRandomRegion()
-		if err != nil {
-			return nil, err
-		}
-		logging.Logger.Infof("Creating session in region - %s", region)
-	}
-	session, err := session.NewSession(&awsgo.Config{
-		Region: awsgo.String(region)},
-	)
-	return session, err
+// TestListS3BucketArgs represents arguments for TestListS3Bucket
+type TestListS3BucketArgs struct {
+	bucketTags  []map[string]string
+	batchSize   int
+	shouldError bool
+	shouldMatch bool
 }
 
-// setupNukeTests sets up common operations for nuke S3 tests.
-func setupNukeTests(t *testing.T) SetupInfo {
-	var s SetupInfo
+// testList3Bucket - helper function for TestListS3Bucket
+func testListS3Bucket(t *testing.T, args TestListS3BucketArgs) {
+	awsParams, err := newAWSParams()
 
-	region, err := getRandomRegion()
 	if err != nil {
-		assert.Fail(t, errors.WithStackTrace(err).Error())
+		assert.Failf(t, "Failed to setup AWS params", errors.WithStackTrace(err).Error())
```

Comment:
Use `require.Fail` or `require.NoError` to cut the test short. Otherwise, execution will continue.

Reply:
Thanks - wasn't aware of that - have used `require.NoError` - removes the if/else check also.